presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -242,7 +242,7 @@ protected void setup(Binder binder)
             });
             configBinder(binder).bindConfigGlobalDefaults(HttpClientConfig.class, config -> {
                 config.setAuthenticationEnabled(true);
-                config.setKerberosPrincipal(kerberosPrincipal);
+                config.setKerberosPrincipal(principal.substituteHostnamePlaceholder().toString());
```

Comment:
should not we use this here also: com/facebook/presto/server/security/KerberosAuthenticator.java:68?